tests/manifests/test_configs_consistency.py

```diff
@@ -0,0 +1,709 @@
+            paths += match_path_in_content(content)
+        return paths
+
+    def mutate_empty_dirs(self, container_spec, workload_spec, mutable_empty_dirs: dict[str, MountedEmptyDir]):
+        pass
+
+
+# A consumer which render-config, so will consume only files prefixed by "readfile " + the render-config input files
+@dataclass(frozen=True)
+class RenderConfigContainerPathConsumer(PathConsumer):
+    inputs_files: dict[str, str] = field(default_factory=dict)
+    env: dict[str, str] = field(default_factory=dict)
+    output: tuple = field(default_factory=tuple)
+
+    @classmethod
+    def from_container_spec(
+        cls, container_spec, workload_spec, templates, other_secrets, mutable_empty_dirs: dict[str, MountedEmptyDir]
+    ):
+        all_mounted_files = get_all_mounted_files(
+            workload_spec, container_spec["name"], templates, other_secrets, mutable_empty_dirs
+        )
+        args = container_spec["args"]
+        for idx, cmd in enumerate(args):
+            if cmd == "-output":
+                target = args[idx + 1]
+                output = ParentMount("/".join(target.split("/")[:-1])), MountNode(target.split("/")[-1])
+                break
+
+        render_config_container = cls(
+            inputs_files={input_file: all_mounted_files[input_file] for input_file in container_spec["args"][3:]},
+            env={e["name"]: e["value"] for e in container_spec.get("env", [])},
+            output=output,
+        )
+
+        return render_config_container
+
+    def mutate_empty_dirs(self, container_spec, workload_spec, mutable_empty_dirs: dict[str, MountedEmptyDir]):
+        # We trim readfile calls from the rendered content
+        for volume_mount in container_spec.get("volumeMounts", []):
+            volume = get_volume_from_mount(workload_spec, volume_mount)
+            if "emptyDir" in volume and volume_mount["mountPath"] == self.output[0].path:
+                assert "subPath" not in volume_mount, "render-config should not target a file mounted using `subPath`"
+                mutable_empty_dirs[volume["name"]].render_config_outputs[self.output[1].node_name] = "\n".join(
+                    [
+                        re.sub(r"{{\s+(?:readfile\s+)(?:^|\s|\".+)\s*}}", "", content)
+                        for content in self.inputs_files.values()
+                    ]
+                )
+
+    def path_is_used_in_content(self, path) -> bool:
+        return (
+            find_keys_mounts_in_content(
+                path,
+                [node_path(*self.output)]
+                + list(self.env.values())
+                + list(self.inputs_files.keys())
+                + list(self.inputs_files.values()),
+            )
+            # for now we deliberately mount too many files in config-templates
+            or path.startswith("/conf")
+            # we also deliberately ignore files which are in the same directory as our output
+            # as we are most certainly cumulating files here
```

Comment:
```suggestion
            # as we are most certainly accumulating files here
```

This does feel a bit strange to me though as we're only writing 1 file out here?